Commands/LeaderboardCommand.cs

```diff
@@ -9,45 +9,40 @@
 
 namespace BrackeysBot.Commands
 {
-    public class LeaderboardDisplayData
-    {
-        public int DisplayIndex { get; set; }
-        public IUserMessage Message { get; set; }
-        public IGuild Guild { get; set; }
-    }
-
     public class LeaderboardCommand : ModuleBase
     {
-        private readonly CommandService commands;
+        private readonly KarmaTable _karmaTable;
+        private readonly SettingsTable _settings;
+        private readonly LeaderboardNavigator _navigator;
 
         private const string LEFT_ARROW = "⬅";
         private const string RIGHT_ARROW = "➡";
 
-        private static Dictionary<ulong, LeaderboardDisplayData> TrackedLeaderboards = new Dictionary<ulong, LeaderboardDisplayData>();
-
-        private static int LeaderboardPagesize => int.Parse(BrackeysBot.Settings["leaderboard-pagesize"]);
-
-        public LeaderboardCommand(CommandService commands)
+        public LeaderboardCommand(KarmaTable karmaTable, SettingsTable settings, LeaderboardNavigator navigator)
         {
-            this.commands = commands;
+            _karmaTable = karmaTable;
+            _settings = settings;
+            _navigator = navigator;
         }
 
         [Command("leaderboard")]
+        [HelpData("leaderboard", "Displays the karma leaderboard.")]
         public async Task DisplayLeaderboard()
         {
-            var builder = await BuildForNextPlaces(0, Context.Guild);
+            int.TryParse(_settings["leaderboard-pagesize"], out int pagesize);
+
+            var builder = await BuildForNextPlaces(_karmaTable, Context.Guild, 0, pagesize);
             var message = await ReplyAsync(string.Empty, false, builder);
 
             await message.AddReactionAsync(new Emoji(LEFT_ARROW));
             await message.AddReactionAsync(new Emoji(RIGHT_ARROW));
 
-            TrackedLeaderboards.Add(message.Id, new LeaderboardDisplayData() { DisplayIndex = 0, Message = message, Guild = Context.Guild });
+            _navigator.AddTrackedMessage(new LeaderboardNavigator.LeaderboardDisplayData() { DisplayIndex = 0, Message = message, Guild = Context.Guild });
         }
 
-        private static async Task<EmbedBuilder> BuildForNextPlaces(int startIndex, IGuild guild)
+        private static async Task<EmbedBuilder> BuildForNextPlaces(KarmaTable table, IGuild guild, int startIndex, int pagesize)
         {
-            KarmaTable table = BrackeysBot.Karma;
-            IEnumerable<Tuple<ulong, int>> places = table.GetLeaderboardPlaces(startIndex, LeaderboardPagesize);
+            IEnumerable<Tuple<ulong, int>> places = table.GetLeaderboardPlaces(startIndex, pagesize);
```

Comment:
Use a dictionary here. I'm assuming the `ulong` stands for the user id and `int` for the number of points.

Reply:
But you can't order a dictionary.
Even if I were to change it, I would have to change it to IEnumerable<KeyValuePair<ulong, int>> after sorting.

Reply:
Forgo about that. I always use this implementation for a generic ordered dictionary: https://github.com/mattmc3/dotmore/blob/master/dotmore/Collections/Generic/OrderedDictionary.cs